### DISCUSSION TITLE
ttrpg-convert unable to launch from MacOS Terminal

### DISCUSSION THREAD

**Discussion by Drown6709** [ebullient/ttrpg-convert-cli#376] [Q&A] [answered]
I've encountered an issue with launching the ttrpg-convert from the MacOS Terminal. It has worked, and I have no clue why it stopped working.

Git commands work no issue. So git clone and git pull works fine, no errors. I've installed homebrew CLI 4.2.12

However, when I try to launch ttrpg-convert I get a bash error: command not found.

iMac:bin bgr$ pwd
/Users/bgr/Documents/010-Documents/DnDRepository/CLI/bin

Mac:bin bgr$ ls
5etools-mirror-1.github.io	ttrpg-convert
dm-source-config.json

iMac:bin bgr$ ttrpg-convert
-bash: ttrpg-convert: command not found

When I launch the ttrpg-convert directly by double clicking it seems to be recognized as a terminal command:

iMac:~ bgr$ /Users/bgr/Documents/010-Documents/DnDRepository/CLI/bin/ttrpg-convert ; exit;
[ 🛑 ERR] Missing required option: '-o=<outputPath>'
Usage: ttrpg-convert [-dhlvV] [--index] [-c=<configPath>] [-g=<datasource>]
                     -o=<outputPath> [] [<input>...] [COMMAND]

  completion  bash/zsh completion:  source <(ttrpg-convert completion)

See 'ttrpg-convert --help' for more information.
logout

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

What am I doing wrong? :(

**Comment by ebullient** [comment]
From what it looks like, you've been following Josh's Windows tutorial very closely.

If you're on MacOS and using homebrew to install/update the ttprg-convert CLI, you don't need to download the CLI manually.

The output, as I see it, is correct: it's telling you the right thing. You're trying to run it without giving it any arguments (the e.g. the output directory to write to, `-o`).

- `ttprg-convert --help`
- `ttprg-convert --version`

If those work, you're good to go.

For reference: [Use Homebrew](https://www.ebullient.dev/projects/ttrpg-convert-cli/docs/alternateRun.html#use-homebrew)

**Comment by Drown6709** [comment]
Thanks for the quick reply!

I think the issue is with the path. Because launching the ttrpg-convert in the terminal without arguments, I’d expect it to run exactly the same as if I double click the exe-file. It should give me an error telling the arguments are missing.

But I get a bash: command not found error. So terminal doesn’t even run the exe-file.

Running the exe-file by double clicking gives this result:

> iMac:~ bgr$ /Users/bgr/Documents/010-Documents/DnDRepository/CLI/bin/ttrpg-convert ; exit;
[ 🛑 ERR] Missing required option: '-o='
Usage: ttrpg-convert [-dhlvV] [--index] [-c=] [-g=]
-o= [] [...] [COMMAND]

> completion bash/zsh completion: source <(ttrpg-convert completion)

> See 'ttrpg-convert --help' for more information.
logout

> Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

> [Process completed]



But executing with a Terminal command gives this error:

> iMac:bin bgr$ ttrpg-convert
-bash: ttrpg-convert: command not found


Does that make sense?

**Reply by Daoenti** [reply]
If it's in the directory you are running it from try adding a ./

So:

./ttrpg-convert

You either have to specify the direct location to it, and ./ says it's in the current directory, or it has to be in your path.

**Reply by ebullient** [reply, marked as answer]
If you use homebrew, it puts the binary on your path for you. 

Homebrew and JBang both do a good job of helping with path management. Using homebrew on the Mac also makes it easier to update.. (if you're on a mac, at this point, I would recommend it).

**Reply by Drown6709** [reply]
Thanks for your quick answers.

Terminal won't accept the ./-leading. Same error. However, typing the full path makes it work, no issues.

I'm installing xcode to try ad install the brew tap and cli to see if I can get that to work. But just typing out the full path from ROOT works. :)

Thanks for your help. It's much appreciated as is the massive work that clearly has gone into creating the ttrpg-convert.

UPDATE: Installing xcode and then tap and the cli works wonders. No need to include exact path. 🥇
